pkg/signing/handlers/rsa/format.go

```diff
@@ -67,6 +68,9 @@ func KeyData(key interface{}) ([]byte, error) {
 }
 
 func PemBlockForKey(priv interface{}, gen ...bool) *pem.Block {
+	// Handling the panic here means that the above write will error and return that error which might be
+	// acceptable by the end user.
+	defer panics.HandlePanic()
```

Comment:
What happend here: in the body there is a panic because a detected inconsistent state.
In the same function there is panic defer which would result in a nil return.

Why not replace the panic directly by a nil return?
What happens with a nil return? The caller expectes to get a pem block, so it will probaply cause in implicit panic because of a nil pointer access up the call tree.

What should be the advantage? There is finally still a panic, but now you don't know the reason anymore.

A panic defer in the same function providing a panic make really no sense. 
- If there is a nil-able return value indicating an invalid state, then we could change the contract and make returning nil a part of it.
- If there is an error return we should return an error.
- If there is no return at all, then it makes definitely no sense to ignore the panic because it directly leads to inconsistent behaviour up the call tree.
  
If at all the panic should be catched at a place where:
- either the call tree can safely be continued (for example a request can just be aborted and continued with the next request)
- the function provides an error. Then we should return the panic as error. Therefore we would need the possbility to convert the panic into an error return. 

Therefore we need a flavor of the defer function taken an error pointer, which is fed by the recovered error in case the panic should be catched.

All other locations make definitely no sense to place such a defer, because it typically leads to worse error behavior.

Reply:
The panic handler would help in logging the issue. A handler here would add some context to the nil writer. Either that or we add something by logging it out. What do you think? The reason for a handler here in my thought is that, for example, a handler could include a prometheus aggregator that tracks errors by emitting some metric.

>What should be the advantage? There is finally still a panic, but now you don't know the reason anymore.

Sure, but the handler supposedly logged the upward panic already. So ideally, there would be two lines of logs catching both panics. That said, I thought I removed all downwards panics, I might have missed this one. :)

Reply:
Remeber that the caller might still throw the panic, they just want some kind of extra information attached to it.